mtp_api/settings/base.py

```diff
@@ -341,6 +342,33 @@
 AWS_IAM_ROLE_ARN = os.environ.get('AWS_IAM_ROLE_ARN', '')
 ANALYTICAL_PLATFORM_BUCKET_PATH = os.environ.get('ANALYTICAL_PLATFORM_BUCKET_PATH', '')
 
+SWAGGER_SETTINGS = {
+    'USE_SESSION_AUTH': False,
+    'SECURITY_DEFINITIONS': {
+        'oauth2_provider': {
+            'type': 'oauth2',
+            'description': 'test',
+            'flow': 'password',
+            'tokenUrl': urljoin(API_URL, '/oauth2/token/'),
+            'authorizationUrl': urljoin(API_URL, '/oauth2/authorize/'),
+            'scopes': {
+                'read': 'Read scope',
+                'write': 'Write scope',
+            }
+        }
+    },
+    'SHOW_REQUEST_HEADERS': True,
+    'SECURITY': [{
+        'password': ['read', 'write']
+    }],
+    'REFETCH_SCHEMA_WITH_AUTH': True,
+    'OAUTH2_CONFIG': {
+        'clientId': os.environ.get('MTP_SWAGGER_CLIENT_ID'),
```

Comment:
Swagger/redoc would be unable to authenticate, so you wouldn't be able to use them to make requests. Since you don't have to authenticate to view the schemas, my guess would be that the rest of the swagger/redoc functionality would be unimpeded. Happy to define a fallback if you want, though that might be confusing as I believe you can only specify one app for this setting

Reply:
specifying only one app is a little bit of a problem since some apis are app-linked… i guess leave it for now.